### DISCUSSION TITLE
Questions for planed bundler/bundling

### DISCUSSION THREAD

**Discussion by jogibear9988** [rome/tools#1403] [Q&A]
I've a few questions when about how rome would work when it gets a bundler.

1.) Will rome only allow bundling of javascript, or will it also support a html file as a start?
2.) How will rome support javascript modules with cylic references?
3.) Will rome bundler support lazy loaded modules? so if I lazy load a javascript module (for example a plugin), but this references now bundled files? (So is it supported to bundle plugins as extra files)

**Comment by sebmck** [comment]
Hey! Bundling is still really early, and mostly only built out to support our internal usage, as well as that of React Native while I was on that team. So my answers will understandably by pretty vague:

> Will rome only allow bundling of javascript, or will it also support a html file as a start?

Yeah I expect us to support bundling non-JavaScript files. It's tricky to architect, but I think Parcel is good precedent here. We will likely be able to do a lot more sophisticated things with our understandings of cross-language boundaries.

> How will rome support javascript modules with cylic references?

Everything is hoisted into the same scope, similar to Rollup.

> Will rome bundler support lazy loaded modules? so if I lazy load a javascript module (for example a plugin), but this references now bundled files? (So is it supported to bundle plugins as extra files)

Yep. That's a requirement for code splitting. The ability to reference modules declared in other bundles. We can also make it work with our top-level scope too.

**Reply by kelset** [reply]
I know it's been quite a while (and congrats on the announcement today 🚀) - I was wondering if there were any further plans for eventual React Native support :) Or is that something you decided not to focus on for the time being?

**Reply by BasixKOR** [reply]
> Everything is hoisted into the same scope, similar to Rollup.

If this is the case, is it possible to see HMR support? I feel like HMR would be hard to accomplish without clear boundaries between modules.

**Reply by leotm** [reply]
also curious on React Native support ^

**Comment by jogibear9988** [comment]
We bundled our project using "polymer-cli" but this took forever and it broke from time to time. So it would be good to see an bundler wich also supports html as a start.
So also scripts included via "<script src=" could be bundled into one file...

> Yep. That's a requirement for code splitting. The ability to reference modules declared in other bundles. We can also make it work with our top-level scope too.

Yes, but it the modules will only be loaded when configured, so the bundling should exclude them and leave them as seperatly loadable. So some references already loaded in the main app need to be removed from them, and some need to be included.

**Comment by likern** [comment]
> Hey! Bundling is still really early, and mostly only built out to support our internal usage, as well as that of React Native while I was on that team. So my answers will understandably by pretty vague:
> 
> > Will rome only allow bundling of javascript, or will it also support a html file as a start?
> 
> Yeah I expect us to support bundling non-JavaScript files. It's tricky to architect, but I think Parcel is good precedent here. We will likely be able to do a lot more sophisticated things with our understandings of cross-language boundaries.
> 
> > How will rome support javascript modules with cylic references?
> 
> Everything is hoisted into the same scope, similar to Rollup.
> 
> > Will rome bundler support lazy loaded modules? so if I lazy load a javascript module (for example a plugin), but this references now bundled files? (So is it supported to bundle plugins as extra files)
> 
> Yep. That's a requirement for code splitting. The ability to reference modules declared in other bundles. We can also make it work with our top-level scope too.

Would be interesting to know how to use this project in / for React Native.
